agentstack/update.py

```diff
@@ -0,0 +1,66 @@
+import os, sys
+import time
+from packaging.version import parse as parse_version, Version
+import importlib.resources
+from pathlib import Path
+import inquirer
+#from agentstack.utils import term_color, get_version, get_package_path
+from agentstack.utils import term_color, get_version
+from agentstack import packaging
+
+def get_package_path(): # TODO get this from utils in upstream branch
+    return importlib.resources.files('agentstack')
+
+ENDPOINT_URL = "https://pypi.org/simple/agentstack/"
+LAST_CHECK_FILENAME = Path.cwd()/".agentstack-last-update"
```

Comment:
I'm not sure if we want to use `Path.cwd()`. When the user runs `agentstack init`, it will create one there. then when the cd into the project, it'll create another file there.

I've also considered eventually building the CLI to work from any subdirectory within the project, as long as somewhere up the file tree exists `agentstack.json`. If we add that feature, there may end up being multiple `.agentstack-last-update` files. 

Alternatively we could consider modifying a file within the package path. It will be overwritten on update/reinstall but that's fine. Thoughts?

Reply:
<img width="256" alt="image" src="https://github.com/user-attachments/assets/65185809-9174-4494-be9a-5877b98b69c5">

Reply:
I've run in to this a couple times thinking about the design of systems that need to know the project path. 

Possible we abstract this up in to a global variable that gets set when the CLI is initialized to share the same path with every interaction instead of passing it down as an argument.

Reply:
definitely agree there is a better way to do this. a global variable could work but only if the dev has a unique venv for each project and im not sure if we can count on that. alternatively traversing up the tree looking for `agentstack.json` would be more consistent but could also be an expensive process. 

i think we break this off into a new issue and we can discuss implementation there :)